src/actions/__tests__/importMarkdown.ts

```diff
@@ -0,0 +1,351 @@
+`)
+  })
+
+  it('should respect indentation when navigating up and down headings', () => {
+    const markdown = `
+# Heading 1
+a
+## Heading 2
+b
+### Heading 3
+c
+## Heading 2 again
+d
+#### Heading 4
+e
+# Heading 1 again
+f
+### Heading 3 again
+g
+`
+    expect(convertMarkdownToText(markdown)).toBe(`
+- Heading 1
+  - a
+  - Heading 2
+    - b
+    - Heading 3
+      - c
+  - Heading 2 again
+    - d
+    - Heading 4
+      - e
+- Heading 1 again
+  - f
+  - Heading 3 again
+    - g
+`)
+  })
+
+  it('should handle nested headings', () => {
+    const markdown = `
+- List Item 1
+- ## List Item 2
+- List Item 3
+`
+    expect(convertMarkdownToText(markdown)).toBe(`
+- List Item 1
+- List Item 2
+- List Item 3
+`)
+  })
+
+  it('should import emphasis', () => {
+    const markdown = `
+*italic*
+_italic_
+**bold**
+__bold__
+***bold italic***
+~~strikethrough~~
+`
+    expect(convertMarkdownToText(markdown)).toBe(`
+- <em>italic</em> <em>italic</em> <strong>bold</strong> <strong>bold</strong> <em><strong>bold italic</strong></em> <del>strikethrough</del>
+`)
+  })
+
+  it('should import lists', () => {
+    const markdown = `
+1. First ordered item
+2. Second ordered item
+   * Unordered sub-list
+* Unordered item
+- Another unordered item
++ Yet another unordered item
+`
+    expect(convertMarkdownToText(markdown)).toBe(`
+- First ordered item
+- Second ordered item
+  - Unordered sub-list
+- Unordered item
+- Another unordered item
+- Yet another unordered item
+`)
+  })
+
+  it('should import links and resolve reference links', () => {
+    const markdown = `
+[Inline link](https://example.com)
+[Inline link with title](https://example.com "Example")
+
+[Reference link][ref]
+
+[ref]: https://example.com
+`
+    expect(convertMarkdownToText(markdown)).toBe(`
+- <a href="https://example.com">Inline link</a> <a href="https://example.com">Inline link with title</a>
+- <a href="https://example.com">Reference link</a>
+`)
+  })
+
+  it('should import images', () => {
+    const markdown = `
+![Alt text](https://example.com/image.jpg)
+
+![Alt text with title](https://example.com/image.jpg "Image title")
+`
+    expect(convertMarkdownToText(markdown)).toBe(`
+- =image
+  - https://example.com/image.jpg
+- =image
+  - https://example.com/image.jpg
+    - Image title
+`)
+  })
+
+  it('should import inline images as html tag', () => {
+    const markdown = `
+Here's an inline image ![Alt text](https://example.com/image.jpg "Image").
+`
+    expect(convertMarkdownToText(markdown)).toBe(`
+- Here&#39;s an inline image <img src="https://example.com/image.jpg" alt="Alt text" title="Image" />.
+`)
+  })
+
+  it('should import code', () => {
+    const markdown = `
+Inline \`code\` has \`back-ticks around\` it.
+
+\`\`\`javascript
+var s = "JavaScript";
+alert(s);
+\`\`\`
+`
+    expect(convertMarkdownToText(markdown)).toBe(`
+- Inline <code>code</code> has <code>back-ticks around</code> it.
+- =code
+  - var s = "JavaScript";&#10;alert(s);
```

Comment:
Meta attributes modify the appearance/behavior of parents, so this should be:

```
- Inline <code>code</code> has <code>back-ticks around</code> it.
- var s = "JavaScript";&#10;alert(s);
  - =code
```

Reply:
This was following the suggesting regarding rendering delays we discussed for images. I'll change it to the proposed format.

Reply:
Yeah, the rendering delay is kind of competing with the fact that meta attributes and their descendants are hidden by default. It would be a major change to render a meta attribute's descendants independently, so I opt to invert the structure.